src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -604,12 +620,34 @@ void ReadPerfSampleIdAll(PerfEventRingBuffer* ring_buffer, const perf_event_head
                                              PERF_SAMPLE_REGS_USER | PERF_SAMPLE_STACK_USER,
                               .sample_regs_user = SAMPLE_REGS_USER_ALL};
 
-  PerfRecordSample res = ConsumeRecordSample(ring_buffer, header, flags);
+  PerfRecordSample res = ConsumeRecordSample(ring_buffer, header, flags, copy_stack_related_data);
 
-  sched_switch_tracepoint sched_wakeup;
-  std::memcpy(&sched_wakeup, res.raw_data.get(), sizeof(sched_switch_tracepoint));
+  sched_switch_tracepoint sched_switch;
+  std::memcpy(&sched_switch, res.raw_data.get(), sizeof(sched_switch_tracepoint));
 
   ring_buffer->SkipRecord(header);
+
+  if (res.ips_size == 0 || !copy_stack_related_data) {
+    return SchedSwitchPerfEvent{
+        .timestamp = res.time,
+        .ordered_stream = PerfEventOrderedStream::FileDescriptor(ring_buffer->GetFileDescriptor()),
+        .data =
+            {
+                .cpu = res.cpu,
+                // As the tracepoint data does not include the pid of the process that the thread
```

Comment:
This comment is now repeated three times in this file, including twice just a few lines from each other. Can we keep only one and in the other two places have a brief sentences referring to that one (e.g., "See `ConsumeSchedSwitchWithOrWithoutStackPerfEvent` for why the pid can be -1.")?